spigot/src/main/java/org/geysermc/floodgate/util/SpigotCommandUtil.java

```diff
@@ -122,11 +123,23 @@ public void kickPlayer(Object player, String message) {
 
     @Override
     public boolean whitelistPlayer(UUID uuid, String username) {
-        return WhitelistUtils.addPlayer(uuid, username);
+        return WhitelistUtils.addPlayer(uuid, username, this);
     }
 
     @Override
     public boolean removePlayerFromWhitelist(UUID uuid, String username) {
-        return WhitelistUtils.removePlayer(uuid, username);
+        return WhitelistUtils.removePlayer(uuid, username, this);
+    }
+
+    public void setWhitelist(Object offlinePlayer, Boolean isWhitelisted) {
+        Runnable runnable = new Runnable() {
+            @Override
+            public void run() {
+                OfflinePlayer player = (OfflinePlayer)offlinePlayer;
+                player.setWhitelisted(isWhitelisted);
+            }
+        };
+
+        versionSpecificMethods.maybeSchedule(runnable);
```

Comment:
Move this to `WhitelistUtils`. And the `isWhitelisted` parameter can be a primitive.

Reply:
Also, the player parameter can and should be type of `OfflinePlayer`. And rename `isWhitelisted` to `whitelist`